Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx

```diff
@@ -92,97 +112,270 @@ function sortDialog(dialogs: DialogInfo[]) {
+    if (link.dialogName != null) {
+      if (link.trigger != null) {
+        selectTo(link.projectId, link.skillId, link.dialogName, `triggers[${link.trigger}]`);
+      } else {
+        navTo(link.projectId, link.skillId, link.dialogName);
+      }
+    }
+  };
+
+  const renderBotHeader = (bot: BotInProject) => {
+    const link: TreeLink = {
+      displayName: bot.name,
+      projectId: currentProjectId,
+      skillId: bot.projectId,
+      isRoot: true,
+      warningContent: botHasWarnings(bot) ? formatMessage('This bot has warnings') : undefined,
+      errorContent: botHasErrors(bot) ? formatMessage('This bot has errors') : undefined,
     };
+
     return (
-      <span ref={props.group?.data.isRoot && addMainDialogRef} role="grid">
+      <span
+        key={bot.name}
+        css={css`
+          margin-top: -6px;
+          width: 100%;
+          label: bot-header;
+        `}
+        role="grid"
+      >
         <TreeItem
-          depth={0}
-          isActive={!props.group!.isCollapsed}
-          isSubItemActive={!!selected}
-          link={props.group!.data}
-          onDelete={onDeleteDialog}
-          onSelect={toggleCollapse}
+          showProps
+          icon={bot.isRemote ? icons.EXTERNAL_SKILL : icons.BOT}
+          isSubItemActive={isLinkEqual(link, selectedLink)}
+          link={link}
+          menu={[{ label: formatMessage('Create/edit skill manifest'), action: () => {} }]}
+          shiftOut={bot.isRemote ? 28 : 0}
         />
       </span>
     );
   };
 
-  function onRenderCell(nestingDepth?: number, item?: any): React.ReactNode {
+  const renderDialogHeader = (skillId: string, dialog: DialogInfo) => {
+    const warningContent = notificationMap[currentProjectId][dialog.id]
+      .filter((diag) => diag.severity === DiagnosticSeverity.Warning)
+      .map((diag) => diag.message)
+      .join(',');
+    const errorContent = notificationMap[currentProjectId][dialog.id]
+      .filter((diag) => diag.severity === DiagnosticSeverity.Error)
+      .map((diag) => diag.message)
+      .join(',');
+
+    const link: TreeLink = {
+      dialogName: dialog.id,
+      displayName: dialog.displayName,
+      isRoot: dialog.isRoot,
+      projectId: currentProjectId,
+      skillId: null,
+      errorContent,
+      warningContent,
+    };
+    return (
+      <span
+        key={dialog.id}
+        ref={dialog.isRoot ? addMainDialogRef : null}
+        css={css`
+          margin-top: -6px;
+          width: 100%;
+          label: dialog-header;
+        `}
+        role="grid"
+      >
+        <TreeItem
+          showProps
+          icon={icons.DIALOG}
+          isSubItemActive={isLinkEqual(link, selectedLink)}
+          link={link}
+          menu={[
+            {
+              label: formatMessage('Remove this dialog'),
+              icon: 'Delete',
+              action: (link) => {
+                onDeleteDialog(link.dialogName ?? '');
```

Comment:
what does it mean to delete a dialog with blank name?

Reply:
That's only there to make type-checking happy; this is inside the `renderDialogHeader` function, so we know that link.dialogName is never going to be undefined here, but TS doesn't necessarily know that.